### PR TITLE
Add option to disable autocomplete in web share dialog

### PR DESCRIPTION
**Pull request by MorrisJobke**
cc @DeepDiver1975 @LukasReschke @felixboehm @cdamken @bboule @Xenopathic 

Please review

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@nickvergessen this has to be added to the new ocs sharee api as well

File: config/config.sample.php [hunk 1]
```diff
@@ -239,6 +239,13 @@
 	)
 ),
 
+/**
+ * This setting changes the autocompletion result of the sharing dialog. If this
+ * is set to false the user needs to enter the full user or group name of the
+ * recipient and autocompletion doesn't work anymore in the share dialog of the
+ * web UI.
+ */
+'webui-sharing-autocompletion.enabled' => true,
```
Comment (nickvergessen) [inline config/config.sample.php]: Is this necessary to be on disc or can this be moved to appconfig?

File: core/ajax/share.php [hunk 2]
```diff
@@ -379,6 +379,15 @@
 					}
 				}
 
+				$sharingAutocompletion = \OC::$server->getConfig()
+					->getSystemValue('webui-sharing-autocompletion.enabled', true);
+
+				if (!$sharingAutocompletion) {
+					$searchTerm = $_GET['search'];
+					$shareWith = array_filter($shareWith, function($user) use ($searchTerm) {
+						return $user['label'] === $searchTerm;
```
Comment (nickvergessen) [inline core/ajax/share.php]: `label` or `value.shareWith`? I think both.

If you type in the uid, I would expect that you get it returned as well and can click it. But with your patch it only works with the display name.

File: config/config.sample.php (hunk 1 above)
Comment (MorrisJobke) [inline config/config.sample.php]: I can also add this as app config.

**Comment by MorrisJobke** [comment]
I forgot to commit something. I will push this once jenkins has finished.

**Pull request closed without merging**